app/src/main/java/com/example/taskmaster/MainActivity.java

```diff
@@ -31,16 +31,11 @@
 
     public String enteredTaskName = null;
     public String enteredTaskPreference = null;
-    private List<Task> tasks;
+    public List<Task> tasks;
     public TaskMasterDatabase database;
+    RecyclerView recyclerView;

```

Comment:
this recyclerView is not available in your second class "LogDataWhenItComesBackCallback"